src/rust/lib/graph-editor/src/component/node/connection.rs

```diff
@@ -0,0 +1,384 @@
+
+            let line         = Line(Var::<f32>::from(1.0) + thickness);
+            let line         = line.rotate(&angle_a);
+            let line_color   = Srgba::new(1.00, 0.565, 0.0, 1.0);
+            let line_colored = line.fill(line_color);
+
+            let glow           = Line(Var::<f32>::from(15.0) * glow);
+            let glow           = glow.rotate(angle_a);
+            let glow_color     = Srgb::new(1.00, 0.565, 0.0);
+            let gradient_start = Srgba::new(glow_color.red,glow_color.green,glow_color.blue,0.0);
+            let gradient_end   = Srgba::new(glow_color.red,glow_color.green,glow_color.blue,0.0);
+            let glow_gradient  = LinearGradient::new()
+                .add(0.0,gradient_start.into_linear())
+                .add(1.0,gradient_end.into_linear());
+            let glow_color     = SdfSampler::new(glow_gradient)
+                .max_distance(15.0)
+                .slope(Slope::Exponent(4.0));
+            let glow_colored   = glow.fill(glow_color);
+
+            (glow_colored + line_colored).into()
+        }
+    }
+}
+
+
+
+// ==============
+// === Events ===
+// ==============
+
+/// Connection events.
+///
+/// `port_move`   should be emitted TO the `Connection`, so it can update its position and shape.
+/// `hover_start` is emitted FROM the `Connection` when a mouse over event is detected.
+/// `hover_end`   is emitted FROM the `Connection` when a mouse over event ends.
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct Events {
+    pub network     : frp::Network,
+    pub port_move   : frp::Source,
+    pub hover_start : frp::Source,
+    pub hover_end   : frp::Source,
+}
+
+
+
+// ==================
+// === Connection ===
+// ==================
+
+/// Shape view for `Connection`.
+///
+/// Connections are center aligned to make some internal computations easier.
+#[derive(Debug,Default,Clone,Copy)]
+pub struct ConnectionView {}
+impl ShapeViewDefinition for ConnectionView {
+    type Shape = shape::Shape;
+    fn new(_shape:&Self::Shape, _scene:&Scene,shape_registry:&ShapeRegistry) -> Self {
+        let shape_system = shape_registry.shape_system(PhantomData::<shape::Shape>);
+        shape_system.shape_system.set_alignment(
+            alignment::HorizontalAlignment::Center,alignment::VerticalAlignment::Center);
+
+        Self {}
+    }
+}
+
+type ConnectionShapeView = component::ShapeView<ConnectionView>;
+
+/// Connections represent a link between an `InputPort` and an `OutputPort`.
+///
+/// Ports are shown as a connecting line going from one port to the other. The position snd shape
+/// of the `Connection` is defined by the start and end ports.
+///
+/// Connections can be created in the IDE by dragging from one port to another. This is currently handled in
+/// the `GraphEditor`, where events are collected and routed to allow dragging from ports,
+/// existence of a partially connected in-creation `Connection`, and finally the connecting of two
+/// ports.
```

Comment:
There should be 2 ways to connect ports - either dragging, or by clicking (mouse down, mouse up) on one port, moving mouse (the connection follows mouse), and clicking on target port. Both should be implemented.